### PR TITLE
Add `tileSize` field for raster and raster-dem tile sources (#946)

### PR DESCRIPTION
**Pull request by teyotan**
## Launch Checklist

Add `tileSize` field for raster and raster-dem tile sources. Solves #946 

Untitled

### Not in this PR

i18n for `Tile Size`

 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by louwers** [comment]
Cool! I will try it out later today.

**Comment by HarelM** [comment]
Can you please add the tile size string to the en.json and run the refresh so I can tag the relevant traslators?

File: src/locales/he/translation.json [hunk 1]
```diff
@@ -162,6 +163,7 @@
   "Add a new source to your style. You can only choose the source type and id at creation time!": "הוספת מקור לסטייל, ביכולתכם לעשות זאת רק בזמן מסויים",
   "TileJSON URL": "כתובת TileJSON",
   "Tile URL": "כתובת אריחים",
+  "Scheme Type": "__STRING_NOT_TRANSLATED__",
```
Comment (HarelM) [inline src/locales/he/translation.json:166]:
```suggestion
  "Scheme Type": "סוג סכמה",
```

File: src/locales/he/translation.json [hunk 2]
```diff
@@ -171,6 +173,7 @@
   "GeoJSON URL": "כתובת GeoJSON",
   "GeoJSON": "GeoJSON",
   "Cluster": "קיבוץ",
+  "Tile Size": "__STRING_NOT_TRANSLATED__",
```
Comment (HarelM) [inline src/locales/he/translation.json:176]:
```suggestion
  "Tile Size": "גודל אריח",
```

File: src/locales/he/translation.json [hunk 3]
```diff
@@ -81,6 +81,7 @@
   "Close modal": "סגירת חלונית",
   "Debug": "דיבאג",
   "Options": "אפשרויות",
+  "<0>Open in OSM</0> &mdash; Opens the current view on openstreetmap.org": "__STRING_NOT_TRANSLATED__",
```
Comment (HarelM) [inline src/locales/he/translation.json:84]:
```suggestion
```

**Comment by HarelM** [comment]
ISO Code	Language	User
de	German	@josxha
en	English	@HarelM
fr	French	@lhapaipai
hr	Hebrew	@HarelM
ja	Japanese	@keichan34
zh	Simplified Chinese	@jieme

Folks, can you help please with the translation?

**Comment by nyurik** [comment]
You may want to use https://translatewiki.net/ -- this is the tool used for tons of apps for translation, including wikipedia

File: src/locales/ja/translation.json [hunk 4]
```diff
@@ -81,6 +81,7 @@
   "Close modal": "モーダルを閉じる",
   "Debug": "デバッグ",
   "Options": "設定",
+  "<0>Open in OSM</0> &mdash; Opens the current view on openstreetmap.org": "__STRING_NOT_TRANSLATED__",
```
Comment (keichan34) [inline src/locales/ja/translation.json:84]:
```suggestion
  "<0>Open in OSM</0> &mdash; Opens the current view on openstreetmap.org": "<0>OSMで開く</0> &mdash; 現在の地図を openstreetmap.org で開く",
```

File: src/locales/ja/translation.json [hunk 5]
```diff
@@ -171,6 +173,7 @@
   "GeoJSON URL": "GeoJSON URL",
   "GeoJSON": "GeoJSON",
   "Cluster": "クラスタ",
+  "Tile Size": "__STRING_NOT_TRANSLATED__",
```
Comment (keichan34) [inline src/locales/ja/translation.json:176]:
```suggestion
  "Tile Size": "タイルサイズ",
```

File: src/locales/ja/translation.json [hunk 6]
```diff
@@ -162,6 +163,7 @@
   "Add a new source to your style. You can only choose the source type and id at creation time!": "スタイルに新規ソースを追加します。注意: 作成時にソースタイプとIDのみを選択できます。",
   "TileJSON URL": "TileJSON URL",
   "Tile URL": "タイルURL",
+  "Scheme Type": "__STRING_NOT_TRANSLATED__",
```
Comment (keichan34) [inline src/locales/ja/translation.json:166]:
```suggestion
  "Scheme Type": "スキーマ",
```

**Comment by HarelM** [comment]
@nyurik I'm familiar with this project, but I'm not familiar with a good integration to github.
If this is something that you would like to push forward feel free to do so. 
I've used crowdin, which works nice, but it's an overhead for developers I belive. 
In any case, you are welcome to improve the current situation.

File: src/locales/fr/translation.json [hunk 7]
```diff
@@ -172,6 +173,7 @@
   "GeoJSON URL": "URL GeoJSON",
   "GeoJSON": "GeoJSON",
   "Cluster": "Cluster",
+  "Tile Size": "__STRING_NOT_TRANSLATED__",
```
Comment (lhapaipai) [inline src/locales/fr/translation.json:176]:
```suggestion
  "Tile Size": "Dimension d'une tuile",
```

File: src/locales/fr/translation.json [hunk 8]
```diff
@@ -163,6 +163,7 @@
   "Add a new source to your style. You can only choose the source type and id at creation time!": "Ajoutez une nouvelle source à votre style. Vous ne pouvez choisir le type et l'ID de la source qu'au moment de la création !",
   "TileJSON URL": "URL TileJSON",
   "Tile URL": "URL de la tuile",
+  "Scheme Type": "__STRING_NOT_TRANSLATED__",
```
Comment (lhapaipai) [inline src/locales/fr/translation.json:166]:
```suggestion
  "Scheme Type": "Type de schéma",
```

File: src/locales/de/translation.json [hunk 9]
```diff
@@ -163,6 +163,7 @@
   "Add a new source to your style. You can only choose the source type and id at creation time!": "Füge eine neue Quelle zu deinem Stil hinzu. Du kannst den Quellentyp und die ID nur bei der Erstellung auswählen!",
   "TileJSON URL": "TileJSON URL",
   "Tile URL": "Kachel-URL",
+  "Scheme Type": "__STRING_NOT_TRANSLATED__",
```
Comment (josxha) [inline src/locales/de/translation.json:166]:
```suggestion
  "Scheme Type": "Schema-Typ",
```

File: src/locales/de/translation.json [hunk 10]
```diff
@@ -172,6 +173,7 @@
   "GeoJSON URL": "GeoJSON URL",
   "GeoJSON": "GeoJSON",
   "Cluster": "Cluster",
+  "Tile Size": "__STRING_NOT_TRANSLATED__",
```
Comment (josxha) [inline src/locales/de/translation.json:176]:
```suggestion
  "Tile Size": "Kachelgröße",
```

File: src/locales/ja/translation.json [hunk 11]
```diff
@@ -81,6 +81,7 @@
   "Close modal": "モーダルを閉じる",
   "Debug": "デバッグ",
   "Options": "設定",
+  "<0>Open in OSM</0> &mdash; Opens the current view on openstreetmap.org": "<0>OSMで開く</0> &mdash; 現在の地図を openstreetmap.org で開く",
```
Comment (HarelM) [inline src/locales/ja/translation.json:84]:
```suggestion
```

File: src/locales/zh/translation.json [hunk 12]
```diff
@@ -81,6 +81,7 @@
   "Close modal": "关闭模态框",
   "Debug": "调试",
   "Options": "选项",
+  "<0>Open in OSM</0> &mdash; Opens the current view on openstreetmap.org": "__STRING_NOT_TRANSLATED__",
```
Comment (HarelM) [inline src/locales/zh/translation.json:84]:
```suggestion
```

**Pull request merged**